src/ast/passes/fold_literals.cpp

```diff
@@ -808,6 +809,19 @@ std::optional<Expression> LiteralFolder::visit(BlockExpr &expr)
   return std::nullopt;
 }
 
+std::optional<Expression> LiteralFolder::visit(TupleAccess &acc)
+{
+  visit(acc.expr);
+  if (auto *tuple = acc.expr.as<Tuple>()) {
+    if (acc.index >= tuple->elems.size()) {
+      // Let semantic analysis issue the error
```

Comment:
Nit: I wouldn’t be overly prescriptive in the comment, because names change and it’ll become dated.  I would just say “later pass”